julia/kernels/upchan_template.cxx

```diff
@@ -484,25 +484,29 @@ cudaEvent_t cuda{{{kernel_name}}}::execute(cudaPipelineState& /*pipestate*/, con
     // Check error codes
     // TODO: Skip this for performance
     CHECK_CUDA_ERROR(cudaStreamSynchronize(device.getStream(cuda_stream_id)));
-    const std::int32_t error_code = *std::max_element((const std::int32_t*)&*info_host.at(gpu_frame_index).begin(),
-                                                      (const std::int32_t*)&*info_host.at(gpu_frame_index).end());
+    const std::int32_t error_code = *std::max_element((const std::int32_t*)&*info_host.begin(),
+                                                      (const std::int32_t*)&*info_host.end());
     if (error_code != 0)
         ERROR("CUDA kernel returned error code cuLaunchKernel: {}", error_code);
 
-    for (std::size_t i = 0; i < info_host.at(gpu_frame_index).size(); ++i)
-        if (info_host.at(gpu_frame_index)[i] != 0)
+    for (std::size_t i = 0; i < info_host.size(); ++i)
+        if (info_host[i] != 0)
             ERROR("cuda{{{kernel_name}}} returned 'info' value {:d} at index {:d} (zero indicates no error)",
-                info_host.at(gpu_frame_index)[i], i);
+                info_host[i], i);
 
     return record_end_event();
 }
 
 void cuda{{{kernel_name}}}::finalize_frame() {
```

Comment:
All other templates remove the `finalize_frame` function. Only the upchannelizer keeps them, and does something that looks important. Is something different happening here, or is the code just structured differently?

Reply:
The upchannelizer is the only one that uses the ring-buffers so far, so it's different; the others will probably need finalize_frame() methods once they get ring-bufferized.  (Previously, they used to have finalize_frame() methods to free metadata, but since the metadata now uses shared_ptrs, that's not necessary any more.)